src/Symfony/Component/String/Utf8String.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+use Symfony\Component\String\Exception\ExceptionInterface;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Hugo Hamon <hugohamon@neuf.fr>
+ *
+ * @throws ExceptionInterface
+ */
+class Utf8String extends AbstractUnicodeString
+{
+    public function __construct(string $string = '')
+    {
+        if ('' !== $string && !preg_match('//u', $string)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        $this->string = $string;
+    }
+
+    public function append(string ...$suffix): AbstractString
+    {
+        $str = clone $this;
+        $str->string .= 1 >= \count($suffix) ? $suffix[0] ?? '' : implode('', $suffix);
+
+        if (!preg_match('//u', $str->string)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        return $str;
+    }
+
+    public function chunk(int $length = 1): array
+    {
+        if (1 > $length) {
+            throw new InvalidArgumentException('The maximum length of each segment must be greater than zero.');
+        }
+
+        if ('' === $this->string) {
+            return [];
+        }
+
+        $rx = '/(';
+        while (65535 < $length) {
+            $rx .= '.{65535}';
+            $length -= 65535;
+        }
+        $rx .= '.{'.$length.'})/us';
+
+        $chunks = [];
+
+        foreach (preg_split($rx, $this->string, -1, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY) as $chunk) {
+            $chunks[] = $str = clone $this;
+            $str->string = $chunk;
+        }
+
+        return $chunks;
+    }
+
+    public function endsWith($suffix): bool
+    {
+        if ($suffix instanceof parent) {
```

Comment:
```suggestion
        if ($suffix instanceof AbstractString) {
```
`parent` is `AbstractUnicodeString`..
(same in many other methods)